### PR TITLE
Xref Party Mode

### PR DESCRIPTION
**Pull request by mynameisfiber**
This PR adds xref party mode which jumbles up the xref to include a healthy mix of good/bad results. This is going to be useful in creating training data for our xref models.

### CODE REVIEW

**Comment by kjacks** [comment]
@mynameisfiber woohoo party 🎈 🎈 🎈

I'm getting this error when I send an xref request with party=true

Screen Shot 2021-02-17 at 18 55 32

File: aleph/views/xref_api.py [hunk 1]
```diff
@@ -53,6 +61,13 @@ def index(collection_id):
     """
     get_index_collection(collection_id, request.authz.READ)
     result = XrefQuery.handle(request, collection_id=collection_id)
+    if get_flag("party", default=False):
```
Comment (pudo) [inline aleph/views/xref_api.py:64]: I'm not convinced we really want to call this ``party`` - that's a bit cryptic: party to what? The name makes sense in our internal discourse but it'll age really badly in 18 months.

File: aleph/search/__init__.py [hunk 2]
```diff
@@ -103,6 +103,11 @@ def get_index(self):
         return xref_index()
 
 
+class XrefPartyQuery(XrefQuery):
```
Comment (pudo) [inline aleph/search/__init__.py:106]: The alternative to subclassing here would be to define an option in ``XrefQuery.SORT_FIELDS`` which makes the sort available via the normal API.

File: aleph/views/xref_api.py [hunk 3]
```diff
@@ -53,6 +61,13 @@ def index(collection_id):
...
+        # In party mode we merge in some random xref pairs and shuffle the
+        # collection. This gives us a good mix of positive and negative matches
+        results_party = XrefPartyQuery.handle(request, collection_id=collection_id)
+        results_merged = result.results + results_party.result
```
Comment (pudo) [inline aleph/views/xref_api.py:68]: This means we're returning twice as many results as requested in the query. I'm a bit concerned this is going to confuse the frontend, but we can try.

**Review by pudo — COMMENTED** [review]
One thing I wonder is if we should try to find a way to exclude pairings from this mode that have already been decided. Since ES has limits on how many clauses you can feed it, this would probably be a post-filter, which in turn would leave the limit/offset parameters a mess and I think that'll cause a bit of chaos on the frontend. Can you think of a method?

File: aleph/views/xref_api.py (hunk 1 above)
Comment (mynameisfiber) [reply aleph/views/xref_api.py:64]: I'll rename it to the more sensibly boring `evaluation_mode`

File: aleph/search/__init__.py (hunk 2 above)
Comment (mynameisfiber) [reply aleph/search/__init__.py:106]: magic. although then i'd have to modify the request options parser so i can create one XrefQuery with the evaluation mode on and another with it off... right?

**Comment by mynameisfiber** [comment]
> I'm getting this error when I send an xref request with party=true

Fixed! Of course I put the typo in while cleaning up variable names just before pushing :disappointed:

File: aleph/views/xref_api.py (hunk 3 above)
Comment (mynameisfiber) [reply aleph/views/xref_api.py]: @kjacks thoughts?

**Comment by kjacks** [comment]
@mynameisfiber @pudo - ok the front end toggle for eval mode should be good to go. In the UI it felt a bit more consistent and easy to understand to call it Sort by: Default / Sort by: Random. And it's behind the tester flag.

**Comment by mynameisfiber** [comment]
@pudo any other critical changes you'd like us to make on this PR?

**Pull request merged**